src/hooks/useErrorMessage.ts

```diff
@@ -0,0 +1,23 @@
+import { useState, useEffect } from "react";
+
+export const useErrorMessage = (message = "", onUnmount?: () => void) => {
+  const [errorMessage, setErrorMessage] = useState(message);
```

Comment:
So I think here, you can change the naming to make it clearer what the function is doing.

For example, the name `message` doesn't indicate that the message can be replaced: a developer has to know the exact implementation of the hook to use it. If instead the property is called `initialMessage`, it reinforces that this is just the first message, and it can be updated later.

`onUnmount` is a good name: it tells exactly what the argument does.